### ISSUE TITLE
Feature request. Say/sayall block when one is not yet completed

### ISSUE THREAD

**Issue by rafale77** [jishi/node-sonos-http-api#467]
@jishi 

So far the recovering the largest group preset first has been pretty reliable. The only time I made the sayall fail to recover is when two sayall were triggered consecutively without enough time between them. The result is a partial recovery, not to the original state but to a state before the first one is recovered.
A couple of suggestions to address this: 
1. Wait until the first sayall recovery is completed before running the second one.
2. Use the state of the first sayall and do not overwrite it as long as it has not been completed.
3. Block the second sayall altogether not allowing a second command unless the command is preset is completed.

**Comment by jishi** [comment]
It's supposed to handle multiple say commands in a row, but would abort the currently running one, and start announcing the new one. It keeps track of all the "zone states" it identified when starting the TTS, and restores the first one (meaning, the state before the first announce). 

Of course, this isn't perfect, and can give erratic behavior if the second TTS is sent in the same time as it tries to restore the first state. 

So basically, I would expect it to already behave as option 2, but with some random glitches. According to your description, this doesn't seem to be the case. Can you test with the `say` command to see if that behaves correctly?

**Comment by rafale77** [comment]
Just wanted to report that the only times I have seen the new sayall recovery has been working well and that the only failures are due to a repeated request sent before the first request is completed. I can confirm that it aborts the previous request but then does not recover the state to the state prior to the first request. I recovers to the state of playing the first TTS file:
For the following sequence:
Playlist(paused) -> TTS 1 -> TTS 2. Aborts TTS1, completes TTS2 then recovers to the TTS 1 file in paused condition.
I have not yet tested the plane say command. Sorry. Will do in the next few days.
